druntime/src/core/checkedint.d

```diff
@@ -802,6 +830,24 @@ ulong mulu()(ulong x, uint y, ref bool overflow)
 pragma(inline, true)
 ulong mulu()(ulong x, ulong y, ref bool overflow)
 {
+    version (D_InlineAsm_X86_64)
```

Comment:
Unfortunately, dmd is not able to inline this code and fails compilable/warn3882.d after merging https://github.com/dlang/dmd/pull/21524 to master:
```
druntime\import\core\checkedint.d(831): Warning: cannot inline function `core.checkedint.mulu!().mulu` ulong mulu()(ulong x, ulong y, ref bool overflow)
```

Reply:
The generated code even looks worse for the asm version (compiled with -O):

useAsm = false:
```
		push	RBP
		mov	RBP,RSP
		sub	RSP,010h
		mov	R8,RDX
		mov	EAX,R8D
		mov	ECX,ESI
		mul	RCX
		mov	-8[RBP],RAX
		cmp	dword ptr -4[RBP],0
		je	L20
		mov	byte ptr [RDI],1
L20:	mov	EAX,-8[RBP]
		mov	RSP,RBP
		pop	RBP
		ret
```
useAsm = true:
```
		push	RBP
		mov	RBP,RSP
		sub	RSP,030h
		mov	-018h[RBP],RDI
		mov	-010h[RBP],ESI
		mov	-8[RBP],EDX
		mov	dword ptr -028h[RBP],0
		mov	byte ptr -024h[RBP],0
		mov	EAX,-8[RBP]
		mul	dword ptr -010h[RBP]
		mov	-028h[RBP],EAX
		setb	byte ptr -024h[RBP]
		mov	AL,-024h[RBP]
		and	AL,1
		mov	RCX,-018h[RBP]
		or	[RCX],AL
		mov	EAX,-028h[RBP]
		mov	RSP,RBP
		pop	RBP
		ret
```
I guess useAsm should just be disabled for dmd.

Reply:
> Unfortunately, dmd is not able to inline this code and fails compilable/warn3882.d after merging https://github.com/dlang/dmd/pull/21524 to master

Hmm - I'd prefer keeping the pragma, to make clear that this *should* be inlined, independent from DMD limitations.

> The generated code even looks worse for the asm version

Hmm, somewhat longer but at least branchless.

> I guess useAsm should just be disabled for dmd.

Heh, DMD is the only compiler where that asm could be beneficial - GDC doesn't support this asm syntax, and LDC uses an LLVM intrinsic.